web/src/app/chat/ChatPage.tsx

```diff
@@ -521,13 +531,9 @@ export function ChatPage({
     debounce,
   });
 
-  const [hasPerformedInitialScroll, setHasPerformedInitialScroll] =
-    useState(false);
-
-  // on new page
-  useEffect(() => {
```

Comment:
note: moved this into the other "on switch" `useEffect` since we need to coordinate the timing of the scroll